File: Python/Daily_Code_Challenge/250825-to-camel-case.py
import re

def to_camel_case(s):

    words = [w for w in re.split(r'[_\-\s]+', s.lower()) if w]

    print(words)

    if not words:
        return ""

    return words[0] + "".join(w.capitalize() for w in words[1:])


to_camel_case("ye old-_-sea  faring_buccaneer_-_with a - peg__leg----and a_parrot_ _named- _squawk")
to_camel_case("FREE cODE cAMP")


'''
camelCase
Given a string, return its camel case version using the following rules:

Words in the string argument are separated by one or more characters from the following set: space ( ), dash (-), or underscore (_). Treat any sequence of these as a word break.
The first word should be all lowercase.
Each subsequent word should start with an uppercase letter, with the rest of it lowercase.
All spaces and separators should be removed.
Run the Tests (Ctrl + Enter)
Reset this lesson
Get Help
Tests
Passed:1. to_camel_case("hello world") should return "helloWorld".
Passed:2. to_camel_case("HELLO WORLD") should return "helloWorld".
Passed:3. to_camel_case("secret agent-X") should return "secretAgentX".
Passed:4. to_camel_case("FREE cODE cAMP") should return "freeCodeCamp".
Passed:5. to_camel_case("ye old-_-sea  faring_buccaneer_-_with a - peg__leg----and a_parrot_ _named- _squawk") should return "yeOldSeaFaringBuccaneerWithAPegLegAndAParrotNamedSquawk".
'''

File: JavaScript/DailyCodeChallenge/250825_toCamelCase.js
function toCamelCase(s) {

  const regex = /[_\-\s]+/gi
  const cleanSentence = s.toLowerCase().split(regex)

  const camelSplit = cleanSentence.map((word,index) => {
    if (index == 0) return word
    return word[0].toUpperCase() + word.slice(1)
  })
  console.log(camelSplit.join(""))
  return camelSplit.join("")
}


toCamelCase("ye old-_-sea  faring_buccaneer_-_with a - peg__leg----and a_parrot_ _named- _squawk")
toCamelCase("HELLO WORLD")

/*
camelCase
Given a string, return its camel case version using the following rules:

Words in the string argument are separated by one or more characters from the following set: space ( ), dash (-), or underscore (_). Treat any sequence of these as a word break.
The first word should be all lowercase.
Each subsequent word should start with an uppercase letter, with the rest of it lowercase.
All spaces and separators should be removed.

Tests
Passed:1. toCamelCase("hello world") should return "helloWorld".
Passed:2. toCamelCase("HELLO WORLD") should return "helloWorld".
Passed:3. toCamelCase("secret agent-X") should return "secretAgentX".
Passed:4. toCamelCase("FREE cODE cAMP") should return "freeCodeCamp".
Passed:5. toCamelCase("ye old-_-sea  faring_buccaneer_-_with a - peg__leg----and a_parrot_ _named- _squawk") should return "yeOldSeaFaringBuccaneerWithAPegLegAndAParrotNamedSquawk".

*/